test/e2e/utils.go

```diff
@@ -495,18 +513,7 @@ func deleteNamespace(ctx context.Context, cl client.Client, timeout time.Duratio
 		}
 	}
 
-	key := types.NamespacedName{
-		Name: ns.Name,
-	}
-
-	err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
-		err := cl.Get(ctx, key, ns)
-		return errors.IsNotFound(err), nil
-	})
-	if err != nil {
-		return fmt.Errorf("timed out waiting for namespace %s to be deleted: %v", ns.Name, err)
-	}
-	return nil
+	return waitForSpecNsDeletion(ctx, cl, timeout, ns.Name)
```

Comment:
@sunjayBhatia having to wait for the ns to delete after every test is time consuming. It's best for tests to use unique ns's and just make sure the ns delete operation does not error.

Reply:
right, ive been trying to get to the bottom of why namespace deletion is taking so long, if it is something our code is doing or just something about the new image

lets just get #320 merged first

Reply:
@sunjayBhatia it takes even longer on regular k8s clusters. It takes a while due to finalizers.